### PR TITLE
Change default ip for dev server from `0.0.0.0` to `*`

### LINKED ISSUE #3732: 🐛 BUG: `wrangler dev` listens only on ipv4 localhost, not ipv6
### What version of `Wrangler` are you using?

2.6.2

### What operating system are you using?

Linux

### Describe the Bug

`wrangler dev` (with `--experimental-local` in my case, not sure if that matters) says that it listens on `localhost:8787`, but when it says this, it is actually listening specifically on `127.0.0.1:8787`. On systems that support ipv6, the hostname `localhost` resolves to two different addresses, `::1` and `127.0.0.1`. `wrangler dev` does not appear to listen on `[::1]:8787`.

This is usually not a problem, as "correct" programs (e.g. web browsers, `curl`, or `workerd`) will try all addresses returned by the DNS query until one of them works. However, Node 18's `fetch()` function _only_ attempts the first address, throwing an exception if that fails. The first address is always `::1`, therefore `fetch("localhost:8787")` always fails in Node.

Arguably both Node and Wrangler are buggy here. Node should be trying all matching addresses, but Wrangler should also be binding to all matching addresses. Fixing the bug on either side solves the specific problem.

(In my case, I am developing a SvelteKit app with a separate Workers-based backend. Although the SvelteKit app is intended to run on Workers itself, it's easiest to develop using the SvelteKit development workflow on Vite, which uses Node. I wanted it to talk to my local `wrangler dev` instance and ran into this issue.)

### PR DESCRIPTION
**Pull request by jspspike**
This provides support for ipv6 with `wrangler dev`

Fixes #3732 

**Author has addressed the following:**

- Tests
  - [ ] Included
  - [ ] Not necessary because:
- Changeset ([Changeset guidelines](https://github.com/cloudflare/workers-sdk/blob/main/CONTRIBUTING.md#changesets))
  - [x] Included
  - [ ] Not necessary because:
- Associated docs
  - [ ] Issue(s)/PR(s):
  - [ ] Not necessary because:

**Note for PR author:**

We want to celebrate and highlight awesome PR review! If you think this PR received a particularly high-caliber review, please assign it the label `highlight pr review` so future reviewers can take inspiration and learn from it.

### CODE REVIEW

**Comment by jspspike** [comment]
@Cherry After discussing with the team we thought it would be best to remove the ipv6 addresses from the printout

That will require a change in miniflare so this PR is blocked until that is released

File: packages/wrangler/src/dev/proxy.ts [hunk 1]
```diff
@@ -299,13 +308,21 @@ export function usePreviewServer({
...
+							accessibleHosts.push("[::]");
+						}
+					}
 					for (const accessibleHost of accessibleHosts) {
 						logger.log(`- ${localProtocol}://${accessibleHost}:${usedPort}`);
 					}
 				});
-				proxy.server.listen(port, ip);
+
+				proxy.server.listen(port, ip === "*" ? "::" : ip);
```
Comment (petebacondarwin) [inline packages/wrangler/src/dev/proxy.ts:325]: Does `::` always work? Can a user configure their computer to prevent IPV6?

**Review by petebacondarwin — APPROVED** [review]
LGTM assuming that listening on `::` always works on all machines.

File: packages/wrangler/src/dev/proxy.ts (hunk 1 above)
Comment (jspspike) [reply packages/wrangler/src/dev/proxy.ts:325]: I'm not 100% sure, looks like you can disable ipv6 on a computer so I'll test manually and see what happens

Comment (jspspike) [reply packages/wrangler/src/dev/proxy.ts:325]: After testing with
```sh
sysctl -w net.ipv6.conf.all.disable_ipv6=1
sysctl -w net.ipv6.conf.default.disable_ipv6=1
```
manually it seems to still be able to listen on ipv4 addresses

**Pull request merged**